package.json

```diff
@@ -80,8 +80,8 @@
     "test:gas": "hardhat test --report-gas",
     "test:coverage": "hardhat coverage --solcoverjs ./.solcover.js",
     "test:unsafe": "TS_NODE_TRANSPILE_ONLY=1 yarn test --no-compile",
-    "prepare": "yarn build",
-    "build": "hardhat compile",
+    "prepare": "TS_NODE_TRANSPILE_ONLY=1 yarn build",
```

Comment:
It feels unsafe to have this as part of the prepare hook

Reply:
I'll pull it out.  I think your prompt on forcing clean instead to solve the out of date typechain issue.